apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/TransactionPool.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Cloud.Spanner.V1.Logging;
+using Google.Protobuf;
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// The TransactionPool works in conjuntion with the Session pool to warm pooled Sessions
+    /// with the transaction semantics it last used (if appropriate).
+    /// When requesting a pooled session, the request can include transactionoptions which
+    /// is then used as a key in the sessionpool.
+    /// </summary>
+    public static class TransactionPool
+    {
+        // Holds transaction state on sessions, including the last created transaction and options used.
+        static readonly ConcurrentDictionary<Session, SessionInfo> s_sessionInfoTable = new ConcurrentDictionary<Session, SessionInfo>();
+
+        // Holds a list of all active transactions with back pointers to the owning session.
```

Comment:
latter. this class is becoming internal (already is in my other branch), so let's not spend a large amount of time on the doccomments (especially since doccoment work is still incoming).